File: DS/C++_Programming/SinglyCL/Program01.cpp
#include <iostream>
using namespace std;

struct node
{
    int data;
    struct node * next;
};

typedef struct node NODE;
typedef struct node * PNODE;

class SinglyCL
{
    private:
        PNODE first;
        PNODE last;
        int iCount;
    
    public:
        SinglyCL()
        {
            this->first = NULL;
            this->last = NULL;
            this->iCount = 0; 
        }

        void InsertFirst(int no)
        {
            PNODE newn = NULL;

            newn = new NODE;

            newn -> data = no;
            newn -> next = NULL;

            if((this->first == NULL) && (this->last == NULL))
            {
                this->first = newn;
                this->last = newn;
            }
            else
            {
                newn -> next = this->first;
                this->first = newn;
            }

            this->last -> next = this->first;
            this->iCount++;
        }

        void InsertLast(int no)
        {}

        void InsertAtPos(int no)
        {}

        void DeleteFirst()
        {}

        void DeleteLast()
        {}

        void DeleteAtPos(int Pos)
        {}

        void Display()
        {}

        int Count()
        {}
};

int main()
{
    SinglyCL obj;

    obj.InsertFirst(51);
    obj.InsertFirst(21);
    obj.InsertFirst(11);

    return 0;
}

File: DS/Java_Programming/SinglyLL/Program01.java
class SinglyLLNode
{
    public int data;
    public SinglyLLNode next;

    public SinglyLLNode(int no)
    {
        data = no;
        next = null;
    }
}

class SinglyLL
{
    private SinglyLLNode first;
    private int iCount;

    public SinglyLL()
    {
        System.out.println("Object is created successfully");
        first = null;
        iCount = 0;
    }

    public void InsertFirst(int no)
    {

    }

    public void InsertLast(int no)
    {

    }

    public void InsertAtPos(int no, int Pos)
    {

    }

    public void DeleteFirst()
    {

    }

    public void DeleteLast()
    {

    }

    public void DeleteAtPos(int Pos)
    {
        
    }

    public void Display()
    {

    }

    public int Count()
    {
        return iCount;
    }
}

class Program01
{
    public static void main(String A[])
    {
        SinglyLL sobj = new SinglyLL();
    }
}